Reduce login latency

rosstimothy opened the pull request:
Reuse the root cluster auth client during the login process to reduce latency.

Closes https://github.com/gravitational/teleport/issues/26712.
Partially addresses https://github.com/gravitational/teleport/issues/26712.

GavinFrazar commented on lib/client/api.go:
```diff
@@ -3968,10 +4023,41 @@ func (tc *TeleportClient) ssoLogin(ctx context.Context, priv *keys.PrivateKey, c
 // keystore (and into the ssh-agent) for future use.
 func (tc *TeleportClient) ActivateKey(ctx context.Context, key *Key) error {
-	ctx, span := tc.Tracer.Start(
+	_, span := tc.Tracer.Start(
```
nit: `ActivateKey` is only called by `ConnectToRootCluster` now, and no longer updates the trusted CAs.

Should we should make `ActivateKey` unexported to prevent future use of it from tsh code?

GavinFrazar reviewed (approved):
LGTM with a few nits. One more that I couldn't add since the lines weren't in the diff: should update `TeleportClient.DeviceLogin` and `TeleportClient.Login` godoc to remove mention of `ActivateKey`, since the former doesn't use it anymore and the latter should advise passing the key to `ConnectToRootCluster`.

rosstimothy committed:
Reduce login latency

Reuse the root cluster auth client during the login process
to reduce latency.

Closes #26712.
Partially addresses #26712.
